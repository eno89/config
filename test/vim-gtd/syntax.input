" Tests for vim-gtd / syntax
" Initial setup  "{{{1

filetype plugin on
syntax enable

function! TestSynstack(lnum, col)
  return map(synstack(a:lnum, a:col), 'synIDattr(v:val, "name")')
endfunction




function s:test_syntax_1()  "{{{1
  Title "INITIAL STATE"

  tabnew
  tabonly!
  setfiletype gtd

  Assert getline(1, '$') ==> [
  \ 'INBOX',
  \ 'NEXT ACTIONS',
  \ 'CALENDAR',
  \ 'PROJECTS',
  \ 'WAITING FOR',
  \ 'SOMEDAY',
  \ 'ARCHIVE',
  \ 'TRASH',
  \ '; vim: filetype=gtd'
  \ ]

  Assert TestSynstack(1, 1) ==> ['gtdSection', 'gtdSectionTitle']
  Assert TestSynstack(2, 1) ==> ['gtdSection', 'gtdSectionTitle']
  Assert TestSynstack(3, 1) ==> ['gtdSection', 'gtdSectionTitle']
  Assert TestSynstack(4, 1) ==> ['gtdSection', 'gtdSectionTitle']
  Assert TestSynstack(5, 1) ==> ['gtdSection', 'gtdSectionTitle']
  Assert TestSynstack(6, 1) ==> ['gtdSection', 'gtdSectionTitle']
  Assert TestSynstack(7, 1) ==> ['gtdSection', 'gtdSectionTitle']
  Assert TestSynstack(8, 1) ==> ['gtdSection', 'gtdSectionTitle']
  Assert TestSynstack(9, 1) ==> ['gtdComment']
endfunction




function s:test_syntax_2()  "{{{1
  Title "TYPICAL CONTENT"

  tabnew
  tabonly!
  setfiletype gtd

  % delete _
  let texts = [
  \ 'INBOX',
  \ '#1 t: e: s: t: title',
  \ "\t2009-08-18T23:58:44",
  \ "\t\there i go (Anthem mix)",
  \ '#2 t: e: s: t: title',
  \ "\t2009-08-18T23:58:44",
  \ "\t\tNEXT STAY (DJ Noriken Remix)",
  \ 'NEXT ACTIONS',
  \ 'CALENDAR',
  \ '#3 t: e: s: t: title',
  \ "\t2009-08-18T23:58:44",
  \ "\t\tJump!",
  \ 'PROJECTS',
  \ 'WAITING FOR',
  \ 'SOMEDAY',
  \ 'ARCHIVE',
  \ 'TRASH',
  \ '; vim: filetype=gtd'
  \ ]
  for i in texts
    put =i
  endfor
  1 delete _

  Assert getline(1, '$') ==> texts

  Assert TestSynstack(1, 1) ==> ['gtdSection', 'gtdSectionTitle']
  Assert TestSynstack(2, 1) ==> ['gtdSection', 'gtdIssue',
  \ 'gtdIssueTitle', 'gtdIssueId']
  Assert TestSynstack(3, 1) ==> ['gtdSection', 'gtdIssue']
  Assert TestSynstack(4, 1) ==> ['gtdSection', 'gtdIssue']
  Assert TestSynstack(5, 1) ==> ['gtdSection', 'gtdIssue',
  \ 'gtdIssueTitle', 'gtdIssueId']
  Assert TestSynstack(6, 1) ==> ['gtdSection', 'gtdIssue']
  Assert TestSynstack(7, 1) ==> ['gtdSection', 'gtdIssue']
  Assert TestSynstack(8, 1) ==> ['gtdSection', 'gtdSectionTitle']
  Assert TestSynstack(9, 1) ==> ['gtdSection', 'gtdSectionTitle']
  Assert TestSynstack(10, 1) ==> ['gtdSection', 'gtdIssue',
  \ 'gtdIssueTitle', 'gtdIssueId']
  Assert TestSynstack(11, 1) ==> ['gtdSection', 'gtdIssue']
  Assert TestSynstack(12, 1) ==> ['gtdSection', 'gtdIssue']
  Assert TestSynstack(13, 1) ==> ['gtdSection', 'gtdSectionTitle']
  Assert TestSynstack(14, 1) ==> ['gtdSection', 'gtdSectionTitle']
  Assert TestSynstack(15, 1) ==> ['gtdSection', 'gtdSectionTitle']
  Assert TestSynstack(16, 1) ==> ['gtdSection', 'gtdSectionTitle']
  Assert TestSynstack(17, 1) ==> ['gtdSection', 'gtdSectionTitle']
  Assert TestSynstack(18, 1) ==> ['gtdComment']

  Assert TestSynstack(2, 3)
  \ ==> ['gtdSection', 'gtdIssue', 'gtdIssueTitle']
  Assert TestSynstack(2, 4)
  \ ==> ['gtdSection', 'gtdIssue', 'gtdIssueTitle', 'gtdIssueTag']
  Assert TestSynstack(2, 5)
  \ ==> ['gtdSection', 'gtdIssue', 'gtdIssueTitle', 'gtdIssueTag']
  Assert TestSynstack(2, 6)
  \ ==> ['gtdSection', 'gtdIssue', 'gtdIssueTitle']
  Assert TestSynstack(2, 15)
  \ ==> ['gtdSection', 'gtdIssue', 'gtdIssueTitle']
  Assert TestSynstack(2, 16)
  \ ==> ['gtdSection', 'gtdIssue', 'gtdIssueTitle']

  Assert TestSynstack(3, 2)
  \ ==> ['gtdSection', 'gtdIssue', 'gtdNoteDatetime']
endfunction




function s:test_syntax_3()  "{{{1
  Title "FOLDING LEVELS"

  tabnew
  tabonly!
  setfiletype gtd

  % delete _
  let texts = [
  \ 'INBOX',
  \ '#1 t: e: s: t: title',
  \ "\t2009-08-18T23:58:44",
  \ "\t\there i go (Anthem mix)",
  \ '#2 t: e: s: t: title',
  \ "\t2009-08-18T23:58:44",
  \ "\t\tNEXT STAY (DJ Noriken Remix)",
  \ 'NEXT ACTIONS',
  \ 'CALENDAR',
  \ '#3 t: e: s: t: title',
  \ "\t2009-08-18T23:58:44",
  \ "\t\tJump!",
  \ 'PROJECTS',
  \ 'WAITING FOR',
  \ 'SOMEDAY',
  \ 'ARCHIVE',
  \ 'TRASH',
  \ '; vim: filetype=gtd'
  \ ]
  for i in texts
    put =i
  endfor
  1 delete _

  Assert getline(1, '$') ==> texts

  Assert foldlevel(1) ==> 1
  Assert foldlevel(2) ==> 2
  Assert foldlevel(3) ==> 2
  Assert foldlevel(4) ==> 2
  Assert foldlevel(5) ==> 2
  Assert foldlevel(6) ==> 2
  Assert foldlevel(7) ==> 2
  Assert foldlevel(8) ==> 0
  Assert foldlevel(9) ==> 1
  Assert foldlevel(10) ==> 2
  Assert foldlevel(11) ==> 2
  Assert foldlevel(12) ==> 2
  Assert foldlevel(13) ==> 0
  Assert foldlevel(14) ==> 0
  Assert foldlevel(15) ==> 0
  Assert foldlevel(16) ==> 0
  Assert foldlevel(17) ==> 0
  Assert foldlevel(18) ==> 0
endfunction




Test  "{{{1

" __END__
" vim: filetype=vim foldmethod=marker
