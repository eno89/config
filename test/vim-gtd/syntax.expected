not found in 'runtimepath': "ftdetect/*.vim"
===== "INITIAL STATE"
TEST: getline(1, '$')  ==>  [ 'INBOX', 'NEXT ACTIONS', 'CALENDAR', 'PROJECTS', 'WAITING FOR', 'SOMEDAY', 'ARCHIVE', 'TRASH', '; vim: filetype=gtd' ] ... ok
TEST: TestSynstack(1, 1)  ==>  ['gtdSection', 'gtdSectionTitle'] ... ok
TEST: TestSynstack(2, 1)  ==>  ['gtdSection', 'gtdSectionTitle'] ... ok
TEST: TestSynstack(3, 1)  ==>  ['gtdSection', 'gtdSectionTitle'] ... ok
TEST: TestSynstack(4, 1)  ==>  ['gtdSection', 'gtdSectionTitle'] ... ok
TEST: TestSynstack(5, 1)  ==>  ['gtdSection', 'gtdSectionTitle'] ... ok
TEST: TestSynstack(6, 1)  ==>  ['gtdSection', 'gtdSectionTitle'] ... ok
TEST: TestSynstack(7, 1)  ==>  ['gtdSection', 'gtdSectionTitle'] ... ok
TEST: TestSynstack(8, 1)  ==>  ['gtdSection', 'gtdSectionTitle'] ... ok
TEST: TestSynstack(9, 1)  ==>  ['gtdComment'] ... ok
Result: 10 / 10

===== "TYPICAL CONTENT"
TEST: getline(1, '$')  ==>  texts ... ok
TEST: TestSynstack(1, 1)  ==>  ['gtdSection', 'gtdSectionTitle'] ... ok
TEST: TestSynstack(2, 1)  ==>  ['gtdSection', 'gtdIssue', 'gtdIssueTitle', 'gtdIssueId'] ... ok
TEST: TestSynstack(3, 1)  ==>  ['gtdSection', 'gtdIssue'] ... ok
TEST: TestSynstack(4, 1)  ==>  ['gtdSection', 'gtdIssue'] ... ok
TEST: TestSynstack(5, 1)  ==>  ['gtdSection', 'gtdIssue'] ... ok
TEST: TestSynstack(6, 1)  ==>  ['gtdSection', 'gtdIssue', 'gtdIssueTitle', 'gtdIssueId'] ... ok
TEST: TestSynstack(7, 1)  ==>  ['gtdSection', 'gtdIssue'] ... ok
TEST: TestSynstack(8, 1)  ==>  ['gtdSection', 'gtdIssue'] ... ok
TEST: TestSynstack(9, 1)  ==>  ['gtdSection', 'gtdIssue'] ... ok
TEST: TestSynstack(10, 1)  ==>  ['gtdSection', 'gtdSectionTitle'] ... ok
TEST: TestSynstack(11, 1)  ==>  ['gtdSection', 'gtdSectionTitle'] ... ok
TEST: TestSynstack(12, 1)  ==>  ['gtdSection', 'gtdIssue', 'gtdIssueTitle', 'gtdIssueId'] ... ok
TEST: TestSynstack(13, 1)  ==>  ['gtdSection', 'gtdIssue'] ... ok
TEST: TestSynstack(14, 1)  ==>  ['gtdSection', 'gtdIssue'] ... ok
TEST: TestSynstack(15, 1)  ==>  ['gtdSection', 'gtdIssue'] ... ok
TEST: TestSynstack(16, 1)  ==>  ['gtdSection', 'gtdSectionTitle'] ... ok
TEST: TestSynstack(17, 1)  ==>  ['gtdSection', 'gtdSectionTitle'] ... ok
TEST: TestSynstack(18, 1)  ==>  ['gtdSection', 'gtdSectionTitle'] ... ok
TEST: TestSynstack(19, 1)  ==>  ['gtdSection', 'gtdSectionTitle'] ... ok
TEST: TestSynstack(20, 1)  ==>  ['gtdSection', 'gtdSectionTitle'] ... ok
TEST: TestSynstack(21, 1)  ==>  ['gtdComment'] ... ok
TEST: TestSynstack(2, 3)  ==>  ['gtdSection', 'gtdIssue', 'gtdIssueTitle'] ... ok
TEST: TestSynstack(2, 4)  ==>  ['gtdSection', 'gtdIssue', 'gtdIssueTitle', 'gtdIssueTag'] ... ok
TEST: TestSynstack(2, 5)  ==>  ['gtdSection', 'gtdIssue', 'gtdIssueTitle', 'gtdIssueTag'] ... ok
TEST: TestSynstack(2, 6)  ==>  ['gtdSection', 'gtdIssue', 'gtdIssueTitle'] ... ok
TEST: TestSynstack(2, 15)  ==>  ['gtdSection', 'gtdIssue', 'gtdIssueTitle'] ... ok
TEST: TestSynstack(2, 16)  ==>  ['gtdSection', 'gtdIssue', 'gtdIssueTitle'] ... ok
TEST: TestSynstack(3, 2)  ==>  ['gtdSection', 'gtdIssue', 'gtdNoteDatetime'] ... ok
Result: 29 / 29

===== "FOLDING LEVELS"
TEST: getline(1, '$')  ==>  texts ... ok
TEST: foldlevel(1)  ==>  1 ... ok
TEST: foldlevel(2)  ==>  2 ... ok
TEST: foldlevel(3)  ==>  2 ... ok
TEST: foldlevel(4)  ==>  2 ... ok
TEST: foldlevel(5)  ==>  2 ... ok
TEST: foldlevel(6)  ==>  2 ... ok
TEST: foldlevel(7)  ==>  2 ... ok
TEST: foldlevel(8)  ==>  2 ... ok
TEST: foldlevel(9)  ==>  2 ... ok
TEST: foldlevel(10)  ==>  0 ... ok
TEST: foldlevel(11)  ==>  1 ... ok
TEST: foldlevel(12)  ==>  2 ... ok
TEST: foldlevel(13)  ==>  2 ... ok
TEST: foldlevel(14)  ==>  2 ... ok
TEST: foldlevel(15)  ==>  2 ... ok
TEST: foldlevel(16)  ==>  0 ... ok
TEST: foldlevel(17)  ==>  0 ... ok
TEST: foldlevel(18)  ==>  0 ... ok
TEST: foldlevel(19)  ==>  0 ... ok
TEST: foldlevel(20)  ==>  0 ... ok
TEST: foldlevel(21)  ==>  0 ... ok
Result: 22 / 22

===== Total
Result: 61 / 61
